src/strategy-management/Toolkit.ts

```diff
@@ -781,93 +781,23 @@ export class Toolkit {
       sellPriceHigh ?? originalStrategy.sellPriceHigh,
       sellBudget ?? originalStrategy.sellBudget
     );
-    const newEncodedStrategy = encodeStrategy(newStrategy);
-
-    // step 3: to avoid changes due to rounding errors, we will override the new encoded strategy with selected values from the old encoded strategy:
-    // - if budget wasn't defined - we will use the old y
-    // - if no price was defined - we will use the old A and B
-    // - if any price was defined - we will reset z to y
-    // - if any budget was defined - will set z according to MarginalPriceOptions
-    // - if marginalPrice is a number - we will use it to calculate z
-    const encodedBN = encodedStrategyStrToBN(encoded);
-    if (buyBudget === undefined) {
-      newEncodedStrategy.order1.y = encodedBN.order1.y;
-    }
-    if (sellBudget === undefined) {
-      newEncodedStrategy.order0.y = encodedBN.order0.y;
-    }
-
-    if (buyPriceLow === undefined && buyPriceHigh === undefined) {
-      newEncodedStrategy.order1.A = encodedBN.order1.A;
-      newEncodedStrategy.order1.B = encodedBN.order1.B;
-    }
-
-    if (sellPriceLow === undefined && sellPriceHigh === undefined) {
-      newEncodedStrategy.order0.A = encodedBN.order0.A;
-      newEncodedStrategy.order0.B = encodedBN.order0.B;
-    }
-
-    if (buyBudget !== undefined) {
-      if (
-        buyMarginalPrice === undefined ||
-        buyMarginalPrice === MarginalPriceOptions.reset ||
-        encodedBN.order1.y.isZero()
-      ) {
-        newEncodedStrategy.order1.z = newEncodedStrategy.order1.y;
-      } else if (buyMarginalPrice === MarginalPriceOptions.maintain) {
-        // maintain the current ratio of y/z
-        newEncodedStrategy.order1.z = mulDiv(
-          encodedBN.order1.z,
-          newEncodedStrategy.order1.y,
-          encodedBN.order1.y
-        );
-      }
-    }
 
-    if (sellBudget !== undefined) {
-      if (
-        sellMarginalPrice === undefined ||
-        sellMarginalPrice === MarginalPriceOptions.reset ||
-        encodedBN.order0.y.isZero()
-      ) {
-        newEncodedStrategy.order0.z = newEncodedStrategy.order0.y;
-      } else if (sellMarginalPrice === MarginalPriceOptions.maintain) {
-        // maintain the current ratio of y/z
-        newEncodedStrategy.order0.z = mulDiv(
-          encodedBN.order0.z,
-          newEncodedStrategy.order0.y,
-          encodedBN.order0.y
-        );
+    for (const [oldOrder, newOrder] of [
+      [decodedOriginal.order0, newStrategy.order0],
+      [decodedOriginal.order1, newStrategy.order1]
+    ]) {
```

Comment:
The code below should be inside `if (marginalPrice === MarginalPriceOptions.maintain)`, where:
- `marginalPrice` is `sellMarginalPrice` for the first pair of orders
- `marginalPrice` is `buyMarginalPrice` for the second pair of orders